Packages/io.chainsafe.web3-unity/Runtime/GUI/GuiOverlayManager.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Pool;
+
+namespace ChainSafe.Gaming.GUI
+{
+    public class GuiOverlayManager : MonoBehaviour
+    {
+        public GuiScreenFactory screenFactory;
+        public Transform container;
+
+        private readonly ObjectPool<GuiInfoOverlay> pool;
+        private readonly List<GuiInfoOverlay> activeOverlays = new();
+        
+        private int overlayCounter = 1000; // offset to detect when default integer value is sent to one of the methods
+
+        public GuiOverlayManager()
+        {
+            pool = new ObjectPool<GuiInfoOverlay>(CreateOverlay, InitOverlay, ReleaseOverlay, defaultCapacity: 2);
```

Comment:
why so?

Reply:
Because Unity also advises you not to do that. You should do it in awake or start

Reply:
you can do it in awake

Reply:
you can technically use parameter-less constructors since that doesn't cause any reflection issues but they'll be called multiple times in editor due to how scene serialization works in unity plus your referencing could go wrong since most of the serialized instances on the `MonoBehaviour` aren't initialized till awake plus uniformity, using constructors in some while using Awake in others